test/routes/src/server/routes/t_github-auth.js

```diff
@@ -0,0 +1,166 @@
+import Express from 'express';
+import supertest from 'supertest';
+import nock from 'nock';
+import expect from 'expect';
+import { stub } from 'sinon';
+import proxyquire from 'proxyquire';
+
+import { conf } from '../../../../../src/server/helpers/config.js';
+conf.set('GITHUB_AUTH_CLIENT_ID', 'akdjhoizxuclakjsdh');
+conf.set('GITHUB_AUTH_CLIENT_SECRET', 'oiwuowiruncoiuihq3e');
+
+const GITHUB_AUTH_LOGIN_URL = 'http://127.0.0.1/login/url';
+const GITHUB_AUTH_CLIENT_ID = 'example_client_id';
+const GITHUB_AUTH_REDIRECT_URL = 'http://127.0.0.1/redirect/url';
+const GITHUB_AUTH_VERIFY_URL = 'http://127.0.0.1/verify/url';
+const GITHUB_AUTH_CLIENT_SECRET = 'example_client_secret';
+
+const mockConfigGetter = stub();
+mockConfigGetter.withArgs('GITHUB_AUTH_LOGIN_URL').returns(GITHUB_AUTH_LOGIN_URL);
+mockConfigGetter.withArgs('GITHUB_AUTH_CLIENT_ID').returns(GITHUB_AUTH_CLIENT_ID);
+mockConfigGetter.withArgs('GITHUB_AUTH_REDIRECT_URL').returns(GITHUB_AUTH_REDIRECT_URL);
+mockConfigGetter.withArgs('GITHUB_AUTH_VERIFY_URL').returns(GITHUB_AUTH_VERIFY_URL);
+mockConfigGetter.withArgs('GITHUB_AUTH_CLIENT_SECRET').returns(GITHUB_AUTH_CLIENT_SECRET);
+
+const login = proxyquire(
+  '../../../../../src/server/routes/github-auth',
+  {
+    '../helpers/config': {
+      conf: { get: mockConfigGetter }
+    }
+  }
+).default;
+
+describe('The login route', () => {
+  let app;
+
+  beforeEach(() => {
+    app = Express();
+    app.use((req, res, next) => {
+      req.session = {
+        sharedSecret: 'bar'
+      };
+      next();
+    });
+    app.use(login);
+    app.use((err, req, res, next) => {
+      res.status(500).send();
+      next();
+    });
+  });
+
+  describe('authenticate action', () => {
+    it('should redirect to configured auth login page', (done) => {
+      supertest(app)
+        .get('/auth/authenticate')
+        .expect('location', new RegExp(GITHUB_AUTH_LOGIN_URL))
+        .expect(302)
+        .end(done);
+    });
+
+    it('should supply configured GitHub client ID', (done) => {
+      supertest(app)
+        .get('/auth/authenticate')
+        .expect('location', new RegExp(GITHUB_AUTH_CLIENT_ID))
+        .end(done);
+    });
+
+    it('should supply configured redirect URL', (done) => {
+      supertest(app)
+        .get('/auth/authenticate')
+        .expect('location', new RegExp(GITHUB_AUTH_REDIRECT_URL))
+        .end(done);
+    });
+
+    it('should supply a shared secret', (done) => {
+      supertest(app)
+        .get('/auth/authenticate')
+        .expect('location', /state/)
+        .end(done);
+    });
```

Comment:
Yep, question is how deeply we actually need to test it. I guess at this point we are good as it is. Especially that we are not programming random generator just relying on what `crypto` provides us with.